spec/GrumPHP/Formatter/PhpcsFormatterSpec.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace spec\GrumPHP\Formatter;
+
+use GrumPHP\Collection\ProcessArgumentsCollection;
+use GrumPHP\Formatter\PhpcsFormatter;
+use GrumPHP\Process\ProcessBuilder;
+use PhpSpec\ObjectBehavior;
+use Symfony\Component\Process\Process;
+
+/**
+ * @mixin PhpcsFormatter
+ */
+class PhpcsFormatterSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('GrumPHP\Formatter\PhpcsFormatter');
+    }
+
+    function it_is_a_process_formatter()
+    {
+        $this->shouldHaveType('GrumPHP\Formatter\ProcessFormatterInterface');
+    }
+
+    function it_handles_command_exceptions(Process $process)
+    {
+        $process->getOutput()->willReturn('');
+        $process->getErrorOutput()->willReturn('stderr');
+        $this->format($process)->shouldReturn('stderr');
+    }
+
+    function it_handles_invalid_json(Process $process)
+    {
+        $process->getOutput()->willReturn('invalid');
+        $this->format($process)->shouldReturn('invalid');
+    }
+
+    function it_formats_phpcs_json_output_for_single_file(Process $process, ProcessBuilder $processBuilder)
+    {
+        $json = $this->parseJson(array(
+            '/var/www/Classes/Command/CacheCommandController.php' => array('messages' => array(array('fixable' => true),),),
+        ));
+
+        $arguments = new ProcessArgumentsCollection();
+        $process->getOutput()->willReturn($this->getExampleData() . $json);
+        $this->format($process)->shouldBe($this->getExampleData());
+
+        $this->getSuggestedFilesFromJson(json_decode($json, true))->shouldBe(array('/var/www/Classes/Command/CacheCommandController.php'));
+
+        $processBuilder->buildProcess($arguments)->willReturn($process);
+        $process->getCommandLine()->shouldBeCalled();
+        $this->formatErrorMessage($arguments, $processBuilder)
+            ->shouldBe(sprintf(
+                '%sYou can fix some errors by running following command:%s',
+                PHP_EOL . PHP_EOL,
+                PHP_EOL . ''
+            ));
+    }
+
+    function it_formats_phpcs_json_output_for_multiple_files(Process $process, ProcessBuilder $processBuilder)
+    {
+        $json = $this->parseJson(array(
+            '/var/www/Classes/Command/CacheCommandController.php' => array('messages' => array(array('fixable' => true),),),
+            '/var/www/Classes/Command/DebugCommandController.php' => array('messages' => array(array('fixable' => false),),),
+        ));
+
+        $arguments = new ProcessArgumentsCollection(array('phpcbf'));
+        $process->getOutput()->willReturn($this->getExampleData() . $json);
+        $this->format($process)->shouldBe($this->getExampleData());
+
+        $this->getSuggestedFilesFromJson(json_decode($json, true))->shouldBe(array('/var/www/Classes/Command/CacheCommandController.php'));
+
+        $processBuilder->buildProcess($arguments)->willReturn($process);
+        $process->getCommandLine()->willReturn();
```

Comment:
Here you can return a sample commandline which you can check in the `shouldBe() ` of the `formatErrorMessage()` method  on the next line.